### ISSUE TITLE
RDS Instance in AWS Cloudformation

### ISSUE THREAD

**Issue by alexw23** [deis/deis#4812]
Just a feature request to see the automated provisioning of RDS in either a) the cloud formation template, or b) via the aws cli. This would make the stateless install one step easier. It would also mean that all resources are used on a teardown. I was stuck for 15 minutes trying to work out why my stack wouldn't delete (VPC failed to delete) and it turned out to be a security group for RDS still linked up to the VPC in EC2.

It'd also be cool to be able to set a logger (i.e. papertrail, loggly) as configuration setting to have this automatically set the logger config. Just a thought :+1:

**Comment by krancour** [comment]
I can see where this would be an added convenience at the time a cluster is initially provisioned, _but_ one of the primary motivations for running a stateless Deis cluster is that with the state persisted _outside_ the cluster, it's easy to rebuild or replace a cluster without _losing_ state.  If management of the RDS database were coupled to the management of the cluster itself, it becomes impossible to tear down the cluster without also tearing down the database.  Ultimately, that would negate the benefits of running a stateless cluster.

In my own experience managing Deis clusters on AWS (before becoming a Deis maintainer), I tried to always look at my clusters as a layered compositions of CloudFormation stacks.  I created separate CloudFormation templates for managing the VPC (into which I would potentially deploy multiple clusters), and cluster state, which included an RDS database and an S3 bucket for use by the registry.

The VPC stack was "foundational."  Conceptually, other stacks (some unrelated to Deis) sat atop it.  Then for each Deis cluster, I created the "state stack" from the aforementioned template and ran the Deis provisioning scripts to create the stack that manages the Deis cluster itself.  This allowed me to peel back and replace that "topmost" layer at any time without affecting anything "underneath" it (like platform state).

In general, I understand this sort of careful layering to be a good practice with respect to CloudFormation.  I'm not sure I could get on board with explicitly defying that within our provisioning process.
